.docker/api-gateway-nginx.conf

```diff
@@ -0,0 +1,18 @@
+server {
+    server_name localhost;
+    listen 80;
+
+    location = /favicon.ico { access_log off; log_not_found off; }
+    location = /robots.txt { access_log off; log_not_found off; }
+
+    # doesn't do the redirect, but /ping doesn't work:
+    #location /scholarly-articles {
+    # can't be used with proxy_pass because regexes don't make clear what prefix to strip
+    #location ~/scholarly-articles($|/) {
+
+    rewrite ^/scholarly-articles$ /scholarly-articles/ last;
+    location /scholarly-articles/ {
```

Comment:
Prefer stripping of trailing slashes. Does mean that `location` has to be regex though.

Reply:
Oh, just seen the comment...

Reply:
Maybe we can use a regex in `location` with a custom rewrite stripping `/scholarly-articles` inside the block, does that make it too complex? Suggested in https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_pass

Reply:
The tested equivalent with regexes is:
```
    location ~ ^/scholarly-articles($|/) {
        rewrite ^/scholarly-articles/?(.*)? /$1 break;                                                                                                                     
        proxy_pass http://scholarly-articles_web;
        proxy_redirect / /scholarly-articles;
    }
```
It seems a more complex idiom because of the `rewrite`. Preferences?

Reply:
Fits https://docs.google.com/document/d/1JhVpW8dxAuJqnk2YsQa76yxdIV_BlOd1QIi02fIu9mo/edit#heading=h.v2gwhv34xrq, and isn't _much_ more complex.

Reply:
The behavior of the two as they stand is identical, there is no redirect from `/scholarly-articles/` to `/scholarly-articles`. It may be added with an additional `rewrite` maybe.

Reply:
That's fine, the app will do that.

Reply:
Full version:
```
    rewrite ^/scholarly-articles/$ $scheme://$http_host/scholarly-articles permanent;                                                                                      
    location ~ ^/scholarly-articles($|/) {
        rewrite ^/scholarly-articles/?(.*)? /$1 break;
        proxy_pass http://scholarly-articles_web;
        proxy_redirect / /scholarly-articles;
    }   
```